### PR TITLE
[x86/Linux] Use portable JIT helpers

### PR DESCRIPTION
**Pull request by parjong**
This commit enforces the use of portable JIT helpers (instead of platform-specific helpers) for x86/Linux.

### CODE REVIEW

**Comment by parjong** [comment]
\CC @seanshpark

**Comment by janvorli** [comment]
@parjong what is the purpose of this change? I think that the reason why it was not using HCIMPL2 is the __declspecI(naked) that you have effectively removed. My guess is that the __declspecI(naked) is there for performance reasons so that the compiler doesn't create prolog and epilog.

**Comment by parjong** [comment]
@janvorli  In vm/fcall.h, x86 has HCCALL2 macros different from those for other architecture, which results in #8390. After merging #8390, I found  hat those functions becomes incompatible with FnStaticBaseHelper for x86. 
I attempt to resolve these conflicts by using predefined macros as other JIT helper functions already did.

**Comment by janvorli** [comment]
I believe the __declspecI(naked) was there for a performance reason, so we should not get rid of it. @jkotas am I right?

**Comment by jkotas** [comment]
> __declspecI(naked) was there for a performance reason

It is there to make things work. It won't work without it. You can see it in the Windows_NT x86 CI results - pretty much all tests failed.

I think that the right strategy is to put these methods under `#ifdef _MSCVER` and use the portable C++ implementation: Put same ifdefs around the matching lines like `#define JIT_IsInstanceOfClass       JIT_IsInstanceOfClass` in `src/vm/i386/cgencpu.h`.

The portable C++ implementations should work just fine - the current C++ compilers will generate code that performs about as good, and sometimes even better, than handcrafted assembly. We use the portable C++ implementations on most platforms.

File: src/vm/i386/jitinterfacex86.cpp [hunk 1]
```diff
@@ -72,13 +72,37 @@ extern "C" void STDCALL WriteBarrierAssert(BYTE* ptr, Object* obj)
 
 #endif // _DEBUG
 
+#ifdef FEATURE_PAL
+FCDECL3(void, JIT_Stelem_Ref, PtrArray* array, unsigned idx, Object* val)
```
Comment (janvorli) [inline src/vm/i386/jitinterfacex86.cpp]: THere already are portable implementations of these functions in src/vm/jithelpers.cpp. It seems that all you need to do is to ifdef-out for FEATURE_PAL the following defines at the end of the src/vm/i386/cgencpu.h, the portable versions of these functions will automatically kick in.
```C++
#define JIT_ChkCastClass            JIT_ChkCastClass
#define JIT_ChkCastClassSpecial     JIT_ChkCastClassSpecial
#define JIT_IsInstanceOfClass       JIT_IsInstanceOfClass

#define JIT_Stelem_Ref              JIT_Stelem_Ref
```

**Comment by jkotas** [comment]
Also, keep the `__declspec(naked)` functions as they are. They cannot be implemented using FCIMPL macro because this macro does things that do not work well with `__declspec(naked)`

**Comment by jkotas** [comment]
The `__declspec(naked)` methods would need to use `HCIMPL2_RAW` - if you really want to change it.

File: src/vm/i386/jitinterfacex86.cpp [hunk 2]
```diff
@@ -330,6 +331,7 @@ extern "C" __declspec(naked) Object* F_CALL_CONV JIT_ChkCastClassSpecial(MethodT
         jmp             JITutil_ChkCastAny
     }
 }
+#endif // FEATURE_PAL
```
Comment (jkotas) [inline src/vm/i386/jitinterfacex86.cpp]: Nit - the matching ifdef is `_MSCVER`

**Pull request merged**